### ISSUE TITLE
Rule Change: Report `no-underscore-dangle` error when property is created (not used)

### ISSUE THREAD

**Issue by regseb** [eslint/eslint#20162]
### What rule do you want to change?

[`no-underscore-dangle`](https://eslint.org/docs/rules/no-underscore-dangle)

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

[Playground](https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgZm9vID0ge1xuICBfYmFyOiBcImJhelwiLCAvLyAoMSlcbn07XG5jb25zdCBfcXV4ID0gXCJxdXV4XCI7IC8vICgyKVxuY29uc29sZS5sb2coZm9vLl9iYXIpOyAvLyAoMylcbmNvbnNvbGUubG9nKGZvby5fYmFyKTsgLy8gKDQpXG5jb25zb2xlLmxvZyhfcXV4KTsgLy8gKDUpXG5jb25zb2xlLmxvZyhfcXV4KTsgLy8gKDYpIiwib3B0aW9ucyI6eyJydWxlcyI6eyJuby11bmRlcnNjb3JlLWRhbmdsZSI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=):

```js
const foo = {
  _bar: "baz", // (1)
};
const _qux = "quux"; // (2)
console.log(foo._bar); // (3)
console.log(foo._bar); // (4)
console.log(_qux); // (5)
console.log(_qux); // (6)
```

### What does the rule currently do for this code?

1. No error.
2. _Unexpected dangling '\_' in '\_qux'._
3. _Unexpected dangling '\_' in '\_bar'._
4. _Unexpected dangling '\_' in '\_bar'._
5. No error.
6. No error.

### What will the rule do after it's changed?

1. _Unexpected dangling '\_' in '\_bar'._
2. _Unexpected dangling '\_' in '\_qux'._
3. No error.
4. No error.
5. No error.
6. No error.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

The rule reports the error when the variable is created. But for a property, it is when it is used.

- It would be good to harmonize these two behaviors.
- This avoids reporting the error multiple times for the same property.

Options could be added to the rule to allow or prohibit creation or use: `allowCreate`, `allowUse`.

**Comment by kecrily** [comment]
Reporting the error at its original location makes perfect sense, as it aligns with how variables behave. However, new default behavior is clearly a breaking change, so let's wait for others' comments.

**Comment by Tanujkanti4441** [comment]
Hi @regseb, thanks for the issue.

`no-underscore-dangle` rule is feature [frozen](https://eslint.org/docs/latest/contribute/core-rules#frozen-rules) so it won't accept any new changes until it is related to a bug fix.

Similar issue https://github.com/eslint/eslint/issues/17453
